signature-specification.md

```diff
@@ -0,0 +1,230 @@
+   - **`mediaType`**: Defines the media type of signature envelope blob. The supported value is `application/jose+json`
+- **`subject`**: An artifact descriptor referencing to image manifest that is signed.
+- **`annotations`**: This OPTIONAL property contains arbitrary metadata for the artifact manifest. It can be used to store information about the signature.
+
+```
+{
+ "artifactType": "application/vnd.cncf.notary.v2",
+ "blobs": [
+   {
+     "mediaType": "application/jose+json",
+     "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+     "size": 32654
+   }
+ ],
+ "subject": {
+   "mediaType": "application/vnd.oci.image.manifest.v1+json",
+   "digest": "sha256:73c803930ea3ba1e54bc25c2bdc53edd0284c62ed651fe7b00369da519a3c333",
+   "size": 16724
+ },
+ "annotations": {
+   "org.cncf.notary.signature.subject": "wabbit-networks.io"
+ }
+}
+```
+
+### Signature Discovery
+The client should be able to discover all the signatures belonging to an artifact(such as image manifest) by using [ORAS Manifest Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md). ORAS Manifest Referrers API returns a paginated list of all artifacts belonging to a target artifact(such as image manifest). The implementation can filter Notary signature artifacts by either using ORAS Manifest Referrers API or using custom logic on the client. Each Notary signature artifact refers to a signature envelope blob.  
+
+
+## Signature Envelope
+The Signature Envelope is a standard data structure for creating a signed message. A signature envelope consists of the following components:
+* Payload(m): The data that is integrity protected - e.g. container image descriptor.
+* Signed attributes(v): The signature metadata that is integrity protected - e.g. signature expiration time, signing time, etc.
+* Unsigned attributes(u): This OPTIONAL property represents signature metadata that is not integrity protected - e.g. timestamp, certificates, etc.
+* Cryptographic signatures(s): The digital signatures computed on payload and signed attributes.
+
+A signature envelope is `e = {m, v, u, s}` where `s` is signature.
+
+Notary v2 supports [JWS JSON Serialization](https://datatracker.ietf.org/doc/html/rfc7515) as signature envelope format with some additional constraints but is extensible to support any other signature envelope format.
+
+### Payload
+Notary v2 requires Payload to be the [descriptor](https://github.com/opencontainers/image-spec/blob/master/descriptor.md#properties) of the target manifest that is being signed.  
+1. Descriptor MUST contain `mediaType`, `digest`, `size` fields.
+1. Descriptor MAY contain `artifactType` field for artifact manifests, or the `config.mediaType` for `oci.image` based manifests or annotations.
```

Comment:
All optional fields can go to the **signed attribtues**.

Reply:
Technically yes but the idea behind of payload and signed attributes is different. Payload is the actual content we want to sign(integrity protect) while attributes is metadata for the signature or payload and in this case annotation, artifactType or  any other  descriptor data is a payload.